googletest/include/gtest/gtest.h

```diff
@@ -1018,6 +1018,27 @@ class EmptyTestEventListener : public TestEventListener {
   void OnTestProgramEnd(const UnitTest& /*unit_test*/) override {}
 };
 
+class XmlUnitTestResultPrinter : public EmptyTestEventListener {
+ public:
+#if GTEST_HAS_FILE_SYSTEM
+  XmlUnitTestResultPrinter(const char* output_file);
+#endif  // GTEST_HAS_FILE_SYSTEM
+
+  XmlUnitTestResultPrinter(std::ostream* output_stream);
+
+  XmlUnitTestResultPrinter(const XmlUnitTestResultPrinter&) = delete;
+  XmlUnitTestResultPrinter& operator=(const XmlUnitTestResultPrinter&) = delete;
+
+  void OnTestIterationEnd(const UnitTest& unit_test, int iteration) override;
+  void ListTestsMatchingFilter(const std::vector<TestSuite*>& test_suites);
+
+ private:
+#if GTEST_HAS_FILE_SYSTEM
```

Comment:
Do you think we could have a separate interface for "output sink"? 

Perhaps we could even turn `XmlUnitTestResultPrinter` into an abstract class, and have a `FileXmlUnitTestResultPrinter` vs. a `StreamXmlUnitTestResultPrinter`?

Reply:
what do you think about having `XmlUnitTestResultPrinter` be the one that prints to a stream and `FileXmlUnitTestResultPrinter` inheriting from that, opening a file stream to give to the base?

I don't think the additional abstract base, from which both `Stream..` and `File..` inherit is useful here